presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -376,6 +377,20 @@ public void checkCanShowGrants(ConnectorTransactionHandle transaction, Identity
         return grantInfos;
     }
 
+    @Override
+    public void checkCanShowRoles(ConnectorTransactionHandle transactionHandle, Identity identity, String catalogName)
```

Comment:
I think that the answer, rather than adding ``SHOW ROLE GRANTS`` and ``SHOW GRANTS`` to SystemAccessControl, is to make the catalog access control actually work and deny access.